shared/js/ui/base/safari-ui-wrapper.es6.js

```diff
@@ -1,45 +1,58 @@
 /* global safari:false */
+let context
+
+if (safari &&
+        safari.extension &&
+        safari.extension.globalPage &&
+        safari.extension.globalPage.contentWindow) {
+    context = 'popup'
+} else if (safari &&
+        safari.self &&
+        safari.self.tab) {
+    context = 'options'
+} else {
+    throw new Error('safari-ui-wrapper couldn\'t figure out the context it\'s in')
+}
```

Comment:
This wasn't completely necessary, but I figured we might as well keep all the insanity contained in the same place. Hence the changes to the UI models.